src/screens/UserManagement/UserRevamp/DropdownWithLoading.res

```diff
@@ -0,0 +1,160 @@
+                      }
+                      `${activeClasses} font-medium text-start`
+                    }>
+                    <div className="mr-5">
+                      {option.label
+                      ->LogicUtils.snakeToTitle
+                      ->React.string}
+                    </div>
+                    <Tick isSelected={keyValueFromForm === option.value} />
+                  </div>
+                </div>}
+            </Menu.Item>
+          )
+          ->React.array}
+        </div>
+      }}
+    </Menu.Items>
+  }
+}
+
+module DropDownLoading = {
+  @react.component
+  let make = () => {
+    <div className={`${commonDropdownCss} flex justify-center items-center`}>
+      <div className={`flex flex-col text-center items-center animate-spin `}>
+        <Icon name="spinner" size=20 />
+      </div>
+    </div>
+  }
+}
+
+module DropDownNoData = {
+  @react.component
+  let make = () => {
+    <div className={`${commonDropdownCss} flex justify-center items-center`}>
+      <p className="text-semibold text-gray-600 opacity-60">
+        {"No data to display"->React.string}
+      </p>
+    </div>
+  }
+}
+
+@react.component
+let make = (
+  ~options: array<SelectBox.dropdownOption>,
+  ~onClickDropDownApi,
+  ~formKey,
+  ~dropDownLoaderState: dropDownState,
+  ~isRequired=false,
+) => {
+  open LogicUtils
+  let (arrow, setArrow) = React.useState(_ => false)
+
+  let formState: ReactFinalForm.formState = ReactFinalForm.useFormState(
+    ReactFinalForm.useFormSubscription(["values"])->Nullable.make,
+  )
+  let keyValueFromForm = formState.values->getDictFromJsonObject->getString(formKey, "")
+
+  let getNameByLabel = value => {
+    let filteredValueFromForm = options->Array.find(v => v.value === value)
+    switch filteredValueFromForm {
+    | Some(value) => value.label->snakeToTitle
+    | None => "Select a role"
+    }
+  }
+
+  let buttonValue = React.useMemo(() => {
+    switch formState.values->getDictFromJsonObject->Dict.get(formKey) {
+    | Some(value) => getNameByLabel(value->getStringFromJson(""))
+    | None => "Select a role"
+    }
+  }, [keyValueFromForm])
+
+  <Menu \"as"="div" className="relative inline-block text-left p-1">
+    {_menuProps => <>
+      <Menu.Button className="w-full">
+        {_buttonProps => {
+          <div className="w-full flex flex-col">
+            <div
+              className="flex justify-start pt-2 pb-2 text-fs-13 text-jp-gray-900 ml-1 font-semibold">
+              {"Role"->React.string}
+              <RenderIf condition=isRequired>
+                <span className="text-red-950"> {React.string("*")} </span>
+              </RenderIf>
+            </div>
+            <div
+              className="relative inline-flex whitespace-pre leading-5 justify-between text-sm py-3 px-4 font-medium rounded-md hover:bg-opacity-80 bg-white border w-full"
+              onClick={_ => arrow ? () : onClickDropDownApi()->ignore}>
```

Comment:
```suggestion
              onClick={_ => if(!arrow) {onClickDropDownApi()->ignore}}>
```

Reply:
ternary looks cleaner than if in this particular case

Reply:
The point is () we don't need. else case unnecessarily getting added

Reply:
But if looks a lot weirder with a !value

Reply:
Okay 😢